src/helper/directory_helper.rs

```diff
@@ -124,18 +105,19 @@ impl DirectoryHelper {
 
     /// Return the DirectoryListing for the latest version
     pub fn get(&self,
-               directory_key: (&::routing::NameType, u64),
-               versioned    : bool,
-               access_level : &::AccessLevel) -> Result<::directory_listing::DirectoryListing, ::errors::NfsError> {
-        let structured_data = try!(self.get_structured_data(directory_key.0, directory_key.1));
+               directory_id: &::routing::NameType,
+               type_tag    : u64,
+               versioned   : bool,
+               access_level: &::AccessLevel) -> Result<::directory_listing::DirectoryListing, ::errors::NfsError> {
+        let structured_data = try!(self.get_structured_data(directory_id, type_tag));
         if versioned {
            let versions = try!(::safe_client::structured_data_operations::versioned::get_all_versions(&mut *self.client.lock().unwrap(), &structured_data));
            let latest_version = versions.last().unwrap();
-           self.get_by_version(directory_key, access_level, latest_version.clone())
+           self.get_by_version(directory_id, access_level, latest_version.clone())
         } else {
             let private_key = try!(self.client.lock().unwrap().get_public_encryption_key()).clone();
```

Comment:
this should be done like this:
```
            let private_key;
            let secret_key;
            let nonce;

            let encryption_keys = match *access_level {
                ::AccessLevel::Private => {
                    private_key = try!(self.client.lock().unwrap().get_public_encryption_key()).clone();
                    secret_key = try!(self.client.lock().unwrap().get_secret_encryption_key()).clone();
                    nonce = ::directory_listing::DirectoryListing::generate_nonce(directory_key.0);

                    Some((&private_key,
                         &secret_key,
                         &nonce))
                },
                ::AccessLevel::Public => None,
```
because if the client does not have an account (unregistered client) then he will not be able to get encryption keys etc, because he does not have a session packet.